copier/config/user_data.py

```diff
@@ -120,12 +152,20 @@ def query_user_data(
         "str": str,
         "yaml": parse_yaml_string,
     }
-    result: AnyByStrDict = {}
+    env = SandboxedEnvironment(undefined=StrictUndefined, **envops)
```

Comment:
Is there no way to extend `copier.tools.Renderer` for this use case instead of having to do all of this manually here?

Reply:
Doesn't seem like it. The `Renderer` needs a `ConfigData` object which needs `query_user_data()`. Making `query_user_data()` use a `Renderer` would create a logical cyclic dependency :confused:

Reply:
So I took a shot at de-duplicating code between `Renderer` and `query_user_data`, let me know what you think.

Reply:
Continuing discussion in related comments.